src/virtual-select.js

```diff
@@ -1742,6 +1742,12 @@ export class VirtualSelect {
     }
 
     const searchValue = value.replace(/\\/g, '').toLowerCase().trim();
+
+    // Only proceed if the search value actually changed
+    if (searchValue === this.searchValue && !forceSet) {
```

Comment:
Shouldn't this be made under the If above (line 1736), they seams to achieve the same goal.